specification/batch/data-plane/Microsoft.Batch/2017-09-01.6.0/BatchService.json

```diff
@@ -10912,6 +10915,26 @@
       },
       "title": "The execution constraints for a job."
     },
+    "TaskContainerSettings": {
+      "properties": {
+        "containerRunOptions": {
+          "type": "string",
+          "title": "Additional options to the container run command.",
+          "description": "These additional options are supplied as arguments to the \"docker create\" command, in addition to those controlled by the Batch Service."
+        },
+        "imageName": {
```

Comment:
As Manoj's comment: move imageName to top since this is required field.

Reply:
We have it alphabetical now which I think is fine, none of our other objects have the required fields at the top, and it makes no difference in the generate docs or code either way.

Reply:
Are you going with strict alphabetical order globally for listing data elements (fields)?

Reply:
There's not really a pattern to be honest, and I guess I'm not that worried about enforcing one (since all the tooling that consumes the spec doesn't care as far as I know).

Reply:
Agreed, the order of fields doesn't matter in REST, so it's usually not worth the effort. Although note that some companies enforce some kind of order, sometimes arbitrary.